src/modules/Builder/features/ActivityItems/ActivityItems.tsx

```diff
@@ -37,7 +37,9 @@ export const ActivityItems = () => {
   });
 
   const { appletId, activityId, itemId } = useParams();
-  const activeItemIndex = itemId ? items?.findIndex((item) => getEntityKey(item) === itemId) : -1;
+  const activeItemIndex = itemId
+    ? (getValues(itemsName) as ItemFormValues[])?.findIndex((item) => getEntityKey(item) === itemId)
```

Comment:
Should we provide up to date `items` for `activeItem` in row 43 as well, and `items` that are used in `handleAddItem` and `handleInsertItem`?